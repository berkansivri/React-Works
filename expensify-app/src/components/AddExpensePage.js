import React from 'react'

const AddExpensePage = () => (
  <div>
    This is add expense component
  </div>
)

export default AddExpensePage