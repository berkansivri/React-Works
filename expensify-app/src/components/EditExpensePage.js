import React from 'react'

const EditExpensePage = (props) => {
  console.log(props);
  return (
    <div>
      This is edit expense component with id : {props.match.params.id}
    </div>
  )
}

export default EditExpensePage