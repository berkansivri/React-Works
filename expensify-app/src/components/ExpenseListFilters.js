import React from 'react'
import { connect } from 'react-redux'
import { setTextFilter, sortByDate, sortByAmount } from '../actions/filters'
const ExpenseListFilters = (props) => (
  <div>
    <input type="text" 
      value={props.filters.text} 
      onChange={(e) => {
        props.dispatch(setTextFilter(e.target.value))
      }} 
    />
    <select value={props.filters.sortBy}
      onChange={e => {
        e.target.value === "date" ? props.dispatch(sortByDate()) : props.dispatch(sortByAmount())
      }}
    >
      <option value="date">Date</option>
      <option value="amount">Amount</option>
    </select>
  </div>
)

const mapStateToProps = (state) => ({
    filters: state.filters
})

export default connect(mapStateToProps)(ExpenseListFilters)