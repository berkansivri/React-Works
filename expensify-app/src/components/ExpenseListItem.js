import React from 'react'
import { removeExpense } from '../actions/expenses'

const ExpenseListItem = ({ id, description, amount, createdAt, dispatch }) => (
  <div>
    <h3>{description}</h3>
    <p>{amount} - {createdAt}</p>
    <button onClick={() => {
      dispatch(removeExpense({ id }))
    }}>Remove</button>
  </div>
)

export default ExpenseListItem