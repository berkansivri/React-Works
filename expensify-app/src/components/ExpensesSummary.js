import React from 'react'
import getExpensesTotal from '../selectors/expenses-total'
import numeral from 'numeral'
import visibleExpenses from '../selectors/expenses'
import { connect } from 'react-redux'

export const ExpensesSummary = ({ expenseCount, expensesTotal }) => (
  <div>
    <h3>Viewing {expenseCount} expenses totalling  { numeral(expensesTotal / 100).format("$0,0.00") }</h3>
  </div>
)

const mapStateToProps = (state) => {
  const expenses = visibleExpenses(state.expenses, state.filters)
  return {
    expenseCount: expenses.length,
    expensesTotal: getExpensesTotal(expenses)
  }
}

export default connect(mapStateToProps)(ExpensesSummary) 