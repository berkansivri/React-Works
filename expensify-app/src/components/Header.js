import React from 'react'
import { NavLink } from 'react-router-dom'
import { connect } from 'react-redux'
import { startLogout } from '../actions/auth'

export const Header = ({ logout }) => (
  <header>
    <h1>Expensify</h1>
    <NavLink to="/" activeClassName="is-active" exact={true}>Home </NavLink>
    <NavLink to="/dashboard" activeClassName="is-active">Dashboard </NavLink>
    <NavLink to="/create" activeClassName="is-active">Create </NavLink>
    <NavLink to="/help" activeClassName="is-active">Help </NavLink>
    <button onClick={logout}>Logout</button>
  </header>
)

const mapDispatchToProps = (dispatch) => ({
  logout: () => dispatch(startLogout())
})

export default connect(undefined, mapDispatchToProps)(Header)