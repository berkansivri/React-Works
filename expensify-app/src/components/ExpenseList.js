import React from 'react'
import { connect } from 'react-redux'
import ExpenseListItem from './ExpenseListItem'
import selectExpenses from '../selectors/expenses'

const ExpenseList = (props) => (
  <div>
    <h1>Expense List</h1>
    {props.expenses.map(ex => <ExpenseListItem key={ex.id} {...ex} dispatch={props.dispatch}/>)}
  </div>
)

const mapStateToProps = (state) => {
  return {
    expenses: selectExpenses(state.expenses, state.filters)
  }
} 

export default connect(mapStateToProps)(ExpenseList)
