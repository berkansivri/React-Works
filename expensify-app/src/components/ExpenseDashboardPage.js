import React from 'react'
import ExpenseList from './ExpenseList'
import ExpenseListFilters from './ExpenseListFilters'
import ExpensesSummary from './ExpensesSummary'

const Dashboard = () => (
  <div>
    <ExpenseListFilters />
    <ExpensesSummary />
    <ExpenseList />
  </div>
)

export default Dashboard