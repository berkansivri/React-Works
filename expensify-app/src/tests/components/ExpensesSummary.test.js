import React from 'react'
import { shallow } from 'enzyme'
import { ExpensesSummary } from '../../components/ExpensesSummary'

test("should render ExpensesSummary correctly with 1 expense", () => {
  const wrapper = shallow(<ExpensesSummary expenseCount={1} expensesTotal={235} />)
  expect(wrapper).toMatchSnapshot()
})

test("should render ExpensesSummary correctly with multiple expenses", () => {
  const wrapper = shallow(<ExpensesSummary expenseCount={23} expensesTotal={2351245612} />)
  expect(wrapper).toMatchSnapshot()
})