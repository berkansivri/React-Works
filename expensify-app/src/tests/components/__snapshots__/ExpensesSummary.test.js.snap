// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render ExpensesSummary correctly with 1 expense 1`] = `
<div>
  <h3>
    Viewing 
    1
     expenses totalling  
    $2.35
  </h3>
</div>
`;

exports[`should render ExpensesSummary correctly with multiple expenses 1`] = `
<div>
  <h3>
    Viewing 
    23
     expenses totalling  
    $23,512,456.12
  </h3>
</div>
`;
